app/queue.py

```diff
@@ -140,6 +153,42 @@ def __register_scripts(self):
             """
         )
 
+        self.scripts[self.LUA_EXPIRE_INFLIGHTS] = self._redis_client.register_script(
+            """
+            local DEFAULT_CHUNK   = 99
+            local inflight_prefix = ARGV[1]
+            local destination     = ARGV[2]
+            local expire_after    = tonumber(ARGV[3])
+
+            local cursor = "0";
+            local expired_inflights = {}
+            repeat
+                local scan_result = redis.call("SCAN", cursor, "MATCH", inflight_prefix, "COUNT", 100);
+                cursor = scan_result[1]
+                for i, inflight in pairs(scan_result[2]) do
+                    local idle = redis.call("object", "idletime", inflight)
+                    if ( idle > expire_after) then
+                        local count         = tonumber(redis.call("LLEN", inflight))
+                        local chunk_size    = math.min(math.max(0, count-1), DEFAULT_CHUNK)
+                        local current       = 0
+
+                        while current < count do
+                            local elements = redis.call("LRANGE", inflight, 0, chunk_size)
+                            redis.call("LPUSH", destination, unpack(elements))
```

Comment:
@sastels with this `LPUSH` the flights will prioritised over those already queued in the `destination`?? In effect a LIFO??

Reply:
It could be okay, considering that the inflight was already queued up in the inbox before and waited an extra time out to expire?